src/display/lv_display.c

```diff
@@ -396,6 +396,27 @@ void lv_display_set_draw_buffers(lv_display_t * disp, lv_draw_buf_t * buf1, lv_d
     disp->buf_act = disp->buf_1;
 }
 
+void lv_display_set_raw_buffers(lv_display_t * disp, void * buf1, void * buf2, uint32_t buf_size,
+                                lv_display_render_mode_t render_mode)
+{
+    uint32_t w = lv_display_get_horizontal_resolution(disp);
+    uint32_t h = lv_display_get_vertical_resolution(disp);
+    LV_ASSERT(w != 0 && h != 0);
+
+    lv_color_format_t cf = lv_display_get_color_format(disp);
+    uint32_t stride = lv_draw_buf_width_to_stride(w, cf);
+    if(stride * h > buf_size && render_mode != LV_DISPLAY_RENDER_MODE_PARTIAL) {
+        LV_LOG_ERROR("buffer too small for non partial render mode.");
```

Comment:
I think it should be an assert not a fallback because the `flush_cb` for FULL_REFRESH or DIRECT is probably not working with PARTIAL mode.

Reply:
Agree. will update soon.